### ISSUE TITLE
Webhook events are being swallowed and queued indefinitely

### ISSUE THREAD

**Issue by wsams** [argoproj/argo-helm#1216]
### Describe the bug

My organization was using v1.6.4 of this chart and recently upgrade to v1.12.0. We have a webhook event source and a sensor that triggers a workflow to create a ticket in our support application. Recently a customer asked why we hadn't responded to a question, but what had happened was two different people submitted a form that POST'd to this webhook around the same moment. But only one Workflow was executed. I then used `curl` to send more than 2 POST's at the same time but still only one was processed. It's a very serious issue because our customers cannot get reliable production support.

I then re-applied v1.6.4 of this chart and suddenly all of the workflows we thought were going to be sent appeared in the UI and started running. I was also able to send simultaneous POST's with curl and they all processed.

It seems like in some scenario workflows are queued, or the events are queued, I'm not sure. And something about downgrading a chart to argo-events v1.5 works, but not v1.6

### Related helm chart

argo-events, argo-workflows

### To Reproduce

Apply v1.12.0 of the argo-events chart.

Create a webhook event source.

Create a sensor that triggers a workflow.

Set `parallelism = 5` in argo-workflows or using synchronization in the workflow that the sensor triggers.

Run 2 POST's to the webhook simultaneously. Only 1 of the first 2 POST's will trigger a workflow. The other will be queued somewhere.

Apply v1.10.0 of the argo-events chart. This causes many pods to restart including my sensor and at that point the missing workflow will start running.

Run 2 more POST's to the webhook simultaneously. Both of these should start running immediately.

Apply v1.12.0 of the argo-events chart. Run 2 more simultaneous POST's, and observe that these two now behave correctly. Both should run.

If I then delete all of the argo-events event source controller pods, the eventbus controller pods, and the sensor controller pods, and sensor pods, and wait for them to come back, the bug will re-appear. Now only one of many simultaneous POST's run, and the rest are queued in the void.

### Expected behavior

When I install v1.12.0 of the argo-events chart and POST to a webhook more than once simultaneously, all of the POSTs should trigger a workflow. If they go over the parallelism limit then the rest would be queued. And non of them will be queued in the background somewhere.

In addition

### Additional context

Using Kubernetes v1.22.2 and Argo Workflows 3.2.7.

**Comment by wsams** [comment]
I haven't been able to track down what's going on exactly, but seems like something between argo-events 1.5.0 and 1.6.0 is causing this issue. I wasn't sure if this was a chart issue or argo-events issue.

**Comment by wsams** [comment]
As another possible test I applied v1.12.0 of this chart and set the version of argo-events to v1.5.0 using,
```
sensorController:
  tag: v1.5.0
eventsourceController:
  tag: v1.5.0
eventbusController:
  tag: v1.5.0
```
After an additional restart of the controller pods the issue went away. It seems like this may actually be an argo-events issue between v1.5.0 and v1.6.0 than a chart issue. Anyone have any thoughts because I had posted there first and then moved the issue here thinking it was chart related.
